ui/app/components/workspace-switcher.ts

```diff
@@ -0,0 +1,35 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { task, TaskGenerator } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { Workspace } from 'waypoint-pb';
+import ApiService from 'waypoint/services/api';
+
+type Args = {
+  current?: string;
+  route?: string;
+  models?: string[];
+  isOpen?: boolean;
+};
+
+type Result = { workspaces: Workspace.AsObject[] } | { error: Error };
+
+export default class extends Component<Args> {
+  @service api!: ApiService;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+
+    taskFor(this.loadWorkspaces).perform();
+  }
+
+  @task({ restartable: true })
```

Comment:
I think this is the first time I'm seeing `ember-concurrency` and `@task`, super interesting